Jyoon/server structure

creayoon opened the pull request:
ReferenceError: type is not defined

Commit:
lint fix

Commit:
swagger import 성공!

Commit:
gulp json build added

Commit:
와... 성공!
http://localhost:5000/swagger로 swagger-ui 접근 가능!

Commit:
-swagger 및 api-route 업데이트
-user, room controller에 method 생성
-mongo에 crud 생성

postman에서는 잘 되는데 localhost에서는 404 not found 나오는 상황

Commit:
api 수정, user-controller post 수업내용, swagger editor 수정

Commit:
insert 잘되는데 error 메시지 뜨는 상태
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: Can't set headers after they are sent.

Commit:
validCheck() 내에서 에러나는 상태인데 어디서 에러로 빠지는 지 모르겠음
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'validCheck' of undefined

Commit:
http://localhost:5000/api/users?id=www 로
PUT  call하면 postman에서 Cannot PUT /api/users,
GET하면 Server error 뜨는 상태.. 모르겠다..

Commit:
study code

Commit:
test commit

Commit:
type is not defined

Commit:
get은 일단 pass

Commit:
for reverting

Commit:
Revert "for reverting"

This reverts commit 559e6671bc85834cb105898e9666a401e35db6a9.

Commit:
user controller crud fin.

Commit:
code 정리

Commit:
Unexpected token error
mongoDB updateMany 수정중

ovekyc commented on src/server/controllers/user-controller.js:
```diff
@@ -1,46 +1,145 @@
+	static get(req, res, cb) {
+		const { data } = req.query;
+
+		if (type === 'list')
```
이부분에서 에러가 났네요.
line9, 11에서 type이 존재하지 않기때문에 에러가 발생한 듯 보입니다.
이전에 슈도 코드로 작성한 부분인것 같은데, 수정이 필요해 보이네요

ovekyc committed:
Add gitignore

Commit:
mongo updateMany

Commit:
fin. before test

Commit:
post update fin. before test

ovekyc committed:
Edit indent size

ovekyc committed:
[test code] before

ovekyc committed:
Fix test code error

ovekyc committed:
Update mongo-mock module

ovekyc committed:
[test code] after

Commit:
lint fix

Commit:
id field added in user model

Commit:
controller, service, mongo   fin. (electron 재설치 후 재확인 필요)

Commit:
getById fin.

Commit:
update fin. (img null bug)

Commit:
delete

Commit:
delete error case 나눔, mongo delete condition 사용

Commit:
getAll fin.

Commit:
code 정리

Commit:
-room controller, room service created
-sell, book file created
-thinking of making validchk generization, and using valication-controller

Commit:
타입별 valid chk 고민중
